common/base.scm

```diff
@@ -113,10 +113,45 @@
 
 (define *presentations* ($make-hash-table))
 
-(define next-presentation-id (let ((count 0))
-                               (lambda ()
-                                 (set! count (+ count 1))
-                                 count)))
+(define (swank:lookup-presented-object id)
+   "Retrieve the object corresponding to ID.
+The secondary value indicates the absence of an entry."
+   ;; it seems we often get (list 'quote id)
+   (if (list? id)
+     (swank:lookup-presented-object (cadr id))
+     (let* ((nope (list #f))
```

Comment:
I usually do this with a quoted vector, to avoid the new list allocation every time the function is invoked:
```
(let* ((nope '#(nope)) ...) ...)
```